dapps/src/endpoint.rs

```diff
@@ -39,7 +40,7 @@ pub struct EndpointInfo {
 	pub icon_url: String,
 }
 
-pub type Endpoints = Arc<BTreeMap<String, Box<Endpoint>>>;
+pub type Endpoints = Arc<RwLock<BTreeMap<String, Box<Endpoint>>>>;
```

Comment:
Well, the end goal is to push to the BTreeMap new dapp entries as a js developer adds. Currently you have to restart parity to get the new dapp.